Introduce `TypedArgumentConverter` as a more specific `ArgumentConverter` (#2346)
junit-team/junit5: documentation/src/docs/asciidoc/release-notes/release-notes-5.7.0-M2.adoc

  same functionality but a more descriptive name.
* `assertTimeoutPreemptively` in `Assertions` now reports the stacktrace of the timed out
  thread in the cause of the `AssertionFailedError`.

> * New `TypedArgumentConverter` that always convert one specific type to another, therefore
>   reducing boilerplate type checks compared to `ArgumentConverter`.

Reviewer: ```suggestion
* New `TypedArgumentConverter` for converting one specific type to another, therefore
  reducing boilerplate type checks compared to implementing `ArgumentConverter` directly.
```
